src/plugins/esql_datagrid/public/row_viewer.tsx

```diff
@@ -0,0 +1,235 @@
+  );
+
+  const renderDefaultContent = useCallback(
+    () => (
+      <UnifiedDocViewer
+        columns={columns}
+        columnsMeta={columnsMeta}
+        dataView={dataView}
+        hit={actualHit}
+        onAddColumn={addColumn}
+        onRemoveColumn={removeColumn}
+        textBasedHits={hits}
+      />
+    ),
+    [actualHit, addColumn, columns, columnsMeta, dataView, hits, removeColumn]
+  );
+
+  const bodyContent = renderDefaultContent();
+
+  return (
+    <EuiPortal>
+      <EuiFlyoutResizable
+        onClose={onClose}
+        type={flyoutType}
+        size={flyoutWidth}
+        pushMinBreakpoint="xl"
+        data-test-subj="esqlRowDetailsFlyout"
+        onKeyDown={onKeyDown}
+        ownFocus={true}
+        minWidth={minWidth}
+        maxWidth={maxWidth}
+        onResize={setFlyoutWidth}
+        css={{
+          maxWidth: `${isXlScreen ? `calc(100vw - ${DEFAULT_WIDTH}px)` : '90vw'} !important`,
+        }}
+        paddingSize="m"
+      >
+        <EuiFlyoutHeader hasBorder>
+          <EuiFlexGroup
+            direction="row"
+            alignItems="center"
+            gutterSize="m"
+            responsive={false}
+            wrap={true}
+          >
+            <EuiFlexItem grow={false}>
+              <EuiTitle
+                size="xs"
+                data-test-subj="docTableRowDetailsTitle"
+                css={css`
+                  white-space: nowrap;
+                `}
+              >
+                <h2>
+                  {i18n.translate('esqlDataGrid.grid.tableRow.docViewerEsqlDetailHeading', {
+                    defaultMessage: 'Result',
+                  })}
+                </h2>
+              </EuiTitle>
+            </EuiFlexItem>
+            {activePage !== -1 && (
+              <EuiFlexItem data-test-subj={`esqlTableRowNavigation-${activePage}`}>
+                <EuiPagination
+                  aria-label={i18n.translate('esqlDataGrid.grid.flyout.rowNavigation', {
+                    defaultMessage: 'Row navigation',
+                  })}
+                  pageCount={pageCount}
+                  activePage={activePage}
+                  onPageClick={setPage}
+                  compressed
+                  data-test-subj="esqlTableRowNavigation"
+                />
+              </EuiFlexItem>
+            )}
+          </EuiFlexGroup>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody>{bodyContent}</EuiFlyoutBody>
+        <EuiFlyoutFooter>
+          <EuiButtonEmpty
+            iconType="cross"
+            onClick={onClose}
+            flush="left"
+            data-test-subj="esqlRowDetailsFlyoutCloseBtn"
+          >
+            {i18n.translate('esqlDataGrid.grid.flyout.close', {
+              defaultMessage: 'Close',
+            })}
+          </EuiButtonEmpty>
+        </EuiFlyoutFooter>
+      </EuiFlyoutResizable>
+    </EuiPortal>
+  );
+}
+
+// eslint-disable-next-line import/no-default-export
+export default RowViewer;
```

Comment:
This was copied from `src/plugins/discover/public/components/discover_grid_flyout/discover_grid_flyout.tsx`, right? Can we instead make this code sharable outside of discover and reuse in both places? I'm making changes to the DocViewer in https://github.com/elastic/kibana/pull/175787 and it would be great to have the same functionality for ES|QL data grid too in the end.

Reply:
I have already mentioned this above. You have already an issue about it. I would prefer the Discovery team to handle it, you have an issue already here https://github.com/elastic/kibana/issues/167846

Reply:
I am still using the DocViewer, your PR just adds this property in the flyout I see

```
decreaseAvailableHeightBy={80} 
```